### PR TITLE
[BEAM-370] Remove many additional definitions/uses of .named methods

### PR DESCRIPTION
**Pull request by bjchambers**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [*] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [*] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [*] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [*] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by bjchambers** [comment]
R: @dhalperi

**Comment by bjchambers** [comment]
(I'll look into build failures, hold off on review)

**Comment by bjchambers** [comment]
R: @dhalperi 

Everything looks good now.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java [hunk 1]
```diff
@@ -491,24 +471,25 @@ Bound withTestServices(BigQueryServices testServices) {
 
       @Override
       public void validate(PInput input) {
-        if (validate) {
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java]: ?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/package-info.java [hunk 2]
```diff
@@ -25,14 +25,12 @@
...
- *     BigQueryIO.Read.named("Read")
- *                    .from("clouddataflow-readonly:samples.weather_stations");
+ *     BigQueryIO.Read.from("clouddataflow-readonly:samples.weather_stations"));
  * }</pre>
  * and {@code Write} transforms that persist PCollections to external storage:
  * <pre> {@code
  * PCollection<Integer> numbers = ...;
- * numbers.apply(TextIO.Write.named("WriteNumbers")
- *                           .to("gs://my_bucket/path/to/numbers"));
+ * numbers.apply(TextIO.Write.to("gs://my_bucket/path/to/numbers"));
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/package-info.java]: add the labels to apply here?

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java [hunk 3]
```diff
@@ -26,8 +26,8 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Mockito.anyString;
-import static org.mockito.Mockito.eq;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java]: ?
-- 

fine, but ?

Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java]: okay, apparently that made it a header. hilarious.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java (hunk 1 above)
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java]: Not sure about this change -- looks functional. May be correct but in a separate PR?

Comment (bjchambers) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java]: I ended up needing it here. Here is why:

1. We were previously checking that we couldn't apply a BigQueryIO operation without specifying at least a query or table.
2. We were doing so using .named, which we can no longer do.
3. To preserve those tests, I tried ".withoutValidation"
4. This leads to a NPE when constructing the BigQuery read operation, because we *don't* validate that at least one of the table or query is present. Instead, we construct the operation, and then the constructor has a Precondition.checkNotNull that fails.

My understanding is that withoutValidation should prevent the "check with BigQuery to make sure the table/query is correct" (eg., because an earlier stage of the pipeline may produce the table, so we expect it to fail). That doesn't mean we should skip validating the transform is correctly constructed.

TL;DR "NullPointerException: table" is less informative than "You must specify a query or table"

Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java]: +1 LGTM, but could you spin off the BQIO part to a separate PR?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/package-info.java (hunk 2 above)
Comment (bjchambers) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/package-info.java]: I'm not sure we need them here -- we're showing examples of creating transforms. I think the naming of the Write transform distracts from the creation of the IO. Willing to add if you feel strongly they should be here...

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java (hunk 3 above)
Comment (bjchambers) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java]: I guess it happened on organize imports? Not particularly sure why that happened either...

**Comment by dhalperi** [comment]
LGTM, please self-merge
